lnrpc/routerpc/router.proto

```diff
@@ -0,0 +1,96 @@
+syntax = "proto3";
+
+package routerpc;
+
+message PaymentRequest {
+    /**
+    A serialized BOLT-11 payment request that contains all information
+    required to dispatch the payment. If the pay req is invalid, or expired,
+    an error will be returned.
+    */
+    string pay_req = 1;
+
+    /**
+    An absolute limit on the highest fee we should pay when looking for a
+    route to the destination. Routes with fees higher than this will be
+    ignored, if there are no routes with a fee below this amount, an error
+    will be returned.
+    */
+    int64 fee_limit = 2;
+
+    /**
+    An absolute limit on the cumulative CLTV value along the route for this
+    payment. Routes with total CLTV values higher than this will be ignored,
+    if there are no routes with a CLTV value below this amount, an error will
+    be returned.
+    */
+    int32 cltv_limit = 3;
+
+    /**
+    An upper limit on the amount of time we should spend when attempting to
+    fulfill the payment. This is expressed in seconds. If we cannot make a
+    successful payment within this time frame, an error will be returned.
+    */
+    int32 timeout_seconds = 4;
+}
+
+message PaymentResponse {
+    /**
+    The payment hash that we paid to. Provided so callers are able to map
+    responses (which may be streaming) back to their original requests.
+    */
+    bytes pay_hash = 1;
+
+    /**
+    The pre-image of the payment successfully completed.
+    */
+    bytes pre_image = 2;
+
+    /**
+    If not an empty string, then a string representation of the payment error.
+    */
+    string payment_err = 3;
+}
+
+message RouteFeeRequest {
+    /**
+    The destination once wishes to obtain a routing fee quote to.
+    */
+    bytes dest = 1;
+
+    /**
+    The amount one wishes to send to the target destination.
+    */
+    int64 amt_sat = 2;
+}
+message RouteFeeResponse {
+    /**
+    A lower bound of the estimated fee to the target destination within the
+    network.
+    */
+    int64 routing_fee_sat = 1;
```

Comment:
Doesn't it make sense to expose msat fees instead, this way we might miscalculate bc of rounding

Reply:
Well, I guess since it's a lower bound then maybe it's okay?